### ISSUE TITLE
Every device became unavailable after update

### ISSUE THREAD

**Issue by nobges** [krahabb/meross_lan#121]
Hello,

Hope you are doing well.
I recently updated HA and meross_lan, and since then every meross device became unavailable.

First, my config is as follow: I'm running Home Assistant Core (v. 2021.8.0) in a docker container on an openmediavault server. Before the update, my devices were configured using http protocol, with the "device key" field empty.

Then I did the update, and once HA restarted all the devices were greyed out. I tried:
- Checking the logs: there is nothing there, even if I turn on debug tracing (more on that later)
- Rebooting HA, from HA and from docker;
- Adding the API key on existing devices;
- Removing and re-adding devices: popup says configuration was successful, yet the device stays unavailable and it says "configuration failed" when I go to the integration panel. I tried both options: adding them as a new integration manually, and waiting for HA to recognize them automatically;
- Changing the protocol from "auto" to "http" back and forth;
- Turning on debug tracing: when I go to /custom_components/meross_lan/, there is no "traces" folder;
- Controlling the devices through Meross App: it works.

Here is what I didn't try, but am willing to do if there is no other option:
- Hard resetting every device;
- Reinstalling meross_lan from scratch.

Any idea of what I could have done wrong here?

Thanks a lot for your help,
Best regards.

**Comment by nobges** [comment]
Update:

I tried to hard reset one of the device, and reinstall meross_lan from the HACS panel, and I also put some gibberish in the "device key" field and replacing it with the correct information. None of it worked BUT the last one at least gave me something: if I put the wrong key, a new entry appears in logs, stating "Error setting up entry [device_,name] for meross_lan". Once I enter the right key, it goes back to pretending the device was set up properly, and yet:
![Capture d’écran 2021-12-01 à 09 21 18](https://user-images.githubusercontent.com/15885936/144197855-ca949e33-b095-44bd-93a8-971a1c73a41c.png)

**Comment by krahabb** [comment]
Hello @nobges ,
Sorry for that..
Can you see what happens by the HA log? Any critical/warning message should be logged by default so there should be some clues about this with the HA default log settings.
If not, and in order to better inspect the whole issue you could enable debug logging for meross_lan with this entry in configuration.yaml:
```yaml
logger:
  default: info
  logs:
    custom_components.meross_lan: debug
```
Expect a verbose log so just turn it on while trying to manage the issue and then disable it ;)

**Comment by neldoreth** [comment]
Same for me, after update nothing works.

**Comment by krahabb** [comment]
Hello @neldoreth ,
could you please check your HA logs too for any warning or critical error related to meross_lan?

**Comment by krahabb** [comment]
Just checked...the 2.4.x versions (Rush) are not compatible so back in time..(likely working just on 2021.11 and later)
I will fix this asap

**Comment by neldoreth** [comment]
Mine starts working. I don't do nothing spacial. Only switch a couple of times the DND light and now works. 
I am in core-2021.11.5. And now I see you post an update (wow). Thanks a lot. 
I'm gonna update.

**Comment by nobges** [comment]
Hi! Sorry for the late reply.
Your update worked like a charm! Thanks or your responsiveness and hard work!

Out of curiosity, what caused the incompatibility between meross_lan and HA Core?

**Comment by krahabb** [comment]
Nice to see that!
Last version of meross_lan uses a new feature recently added to HA core in order to better categorize entities (https://www.home-assistant.io/blog/2021/11/03/release-202111/#entity-categorization)
This feature is not obviously available on older HA releases so I'd need to check for this and 'prepare for the worst'
I'm trying to keep meross_lan compatible back to HA core 2021.4.0 while still being able to add features and fix bugs.
There was another minor feature compatibility related to some new configuration entities I've introduced for smart thermostats. That too needed some care.

**Comment by krahabb** [comment]
@MiBe17 ,
your issue is likely not related to the same as the one this thread was open for. 
This thread issue has been fixed with latest patch (2.4.2) while yours is likely due to an incorrect key. You can follow #99 and many other threads about this.
